### DISCUSSION TITLE
BigInt to Int in Prisma Schema

### DISCUSSION THREAD

**Discussion by cakctus** [prisma/prisma#17065] [Q&A]
How to change BigInt to Int id in prisma schema? I try do it and I have error
![Screenshot_7](https://user-images.githubusercontent.com/39764706/210063954-0ee0be33-861d-45e3-bc0c-5843fcafbdce.png)

**Comment by nurul3101** [comment]
Hello @cakctus 👋 

Which database are you using?
Did you change the column of a model in your schema file from `BigInt` to `Int` and then invoke `npx prisma migrate dev`?
Did the column which is being converted had existing data in it?

**Reply by cakctus** [reply]
Postgresql, yes i change the column of a model from BigInt to Int and invoke npx prisma migrate dev.
Yes, has existing data

**Reply by nurul3101** [reply]
If the existing data has bigint values which cannot be converted to int values, then you might receive this error.
Can you try creating a new column with int type and then transfer data from the old column(bigint) to the new column(int) instead of directly changing the type of the column?

**Reply by cakctus** [reply]
Sound good, but how can I do this?

**Reply by nurul3101** [reply]
To summarize you would need to follow these steps:

- Add a new column in your model in the `schema.prisma` file which is of type INT and keep the old column of BIGINT type as it is.
- Create a new migration with command `npx prisma migrate dev --create-only`
- Edit the generated migration file to transfer the data from the old column to the new column, the SQL query would look something like this
```
UPDATE my_table SET my_column_int = my_column_bigInt::int;
``` 
- After this run `npx prisma migrate dev` to execute the pending migrations.
- Once the data is transferred to the new column, you can drop the old column and rename the new column to the old column name.

**Reply by cakctus** [reply]
In your expression "my_table" what is ?

**Reply by nurul3101** [reply]
my_table is the name of the table.

**Reply by cakctus** [reply]
I mean my_table is my_column_int or _my_column_bigInt ?

**Reply by nurul3101** [reply]
The value in the my_column_bigInt column is cast to an integer, and the result is stored in the my_column_int column.

The UPDATE statement is used to modify existing records in a table, and the SET clause is used to specify the new values that should be assigned to the columns. In this case, the my_column_int column is being set to the value of my_column_bigInt cast as an integer using the ::int type cast.

**Reply by cakctus** [reply]
in principle it worked but I lost data in the child table that was attached

![Screenshot_10](https://user-images.githubusercontent.com/39764706/210885452-303b3e73-a017-42e6-98b5-3b6ce1c04167.png)

**Reply by nurul3101** [reply]
Can you elaborate on how did you lose the data? Do you mean you lost the data even before dropping the old column?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
